components/org.wso2.carbon.aws/src/main/java/org/wso2/carbon/aws/user/store/mgt/AWSUserStoreManager.java

```diff
@@ -0,0 +1,1858 @@
+        if (hasRoles) {
+            // Add roles to user.
+            addRolesToUser(userName, roleList);
+        }
+    }
+
+    /**
+     * Deletes a user by userName from userstore.
+     *
+     * @param userName of user to delete
+     * @throws UserStoreException exception if any occur.
+     */
+    @Override
+    public void doDeleteUser(String userName) throws UserStoreException {
+
+        String selector = pathToUsers + "/" + userName;
+        if (membershipType.equals(AWSConstants.LINK)) {
+            // List and detach all outgoing typed links from a user object.
+            JSONObject outgoingTypedLinks = awsActions.listOutgoingTypedLinks(null, selector);
+            detachOutgoingTypedLinks(outgoingTypedLinks);
+        } else if (membershipType.equals(AWSConstants.ATTRIBUTE)) {
+            // Remove the user from all role objects.
+            removeUserFromRoles(userName);
+        }
+        // Detach object from parent object.
+        JSONObject detachObject = awsActions.detachObject(userName, pathToUsers);
+        if (detachObject.get(AWSConstants.DETACHED_OBJECT_IDENTIFIER) != null) {
+            String identifier = "$" + detachObject.get(AWSConstants.DETACHED_OBJECT_IDENTIFIER).toString();
+            // Delete object from directory.
+            awsActions.deleteObject(identifier);
+        }
+    }
+
+    /**
+     * Remove a particular user from all role objects.
+     *
+     * @param userName User name.
+     * @throws UserStoreException If any error occur.
+     */
+    protected void removeUserFromRoles(String userName) throws UserStoreException {
+
+        Map<String, String> map = new HashMap<>();
+        List<String> attributes = new LinkedList<>();
+        String nextToken = null;
+        do {
+            JSONObject listChildren = awsActions.listObjectChildren(nextToken, pathToRoles);
+            Object token = listChildren.get(AWSConstants.NEXT_TOKEN);
+            nextToken = (token != null) ? token.toString() : null;
+            Object childrensObj = listChildren.get(AWSConstants.CHILDREN);
+            JSONObject childrens = (childrensObj != null) ? (JSONObject) childrensObj : null;
+            if (childrens != null) {
+                for (Object key : childrens.keySet()) {
+                    String keyValue = pathToRoles + "/" + key.toString();
+                    String existingUsers = getAttributeValue(facetNameOfRole, keyValue, memberOfAttribute);
+                    if (StringUtils.isNotEmpty(existingUsers) && existingUsers.contains(userName)) {
+                        List<String> updatedUserList = new LinkedList<>(Arrays.asList(existingUsers.split(",")));
+                        updatedUserList.remove(userName);
+
+                        map.put(memberOfAttribute, String.join(",", updatedUserList));
+                        attributes.add(AWSConstants.UPDATE_ATTRIBUTES + awsActions.buildPayloadToUpdateObjectAttributes(
```

Comment:
Can't we parse the json object and do this processing than defining constants for json.

Reply:
To update object attribute we need this format of payload [1]. And I'm using this UPDATE_ATTRIBUTES constant in multiple places. here I have done any parsing. Just build the payload as string and pass it to method [2].

[1] https://docs.aws.amazon.com/directoryservice/latest/APIReference/API_UpdateObjectAttributes.html
[2]https://github.com/Biruntha/identity-userstore-aws/blob/f79a97470c42f4e035de8ace1863db73688e6314/components/org.wso2.carbon.aws/src/main/java/org/wso2/carbon/aws/user/store/mgt/util/AWSRestApiActions.java#L406

Reply:
You can convert JSON to string... In this case, what Hasanthi suggest is build the required object as a JSON object first... that way it is much clear to understand the code. With this approach, if you need to make any changes to the object future, it can be done easily.